### ISSUE TITLE
Issue on `bleep compile` in this project

### ISSUE THREAD

**Issue by daddykotex** [oyvindberg/bleep#281]
Installed bleep from [the release page](https://github.com/oyvindberg/bleep/releases/download/v0.0.1-M24/bleep-arm64-apple-darwin.tar.gz)

Following the contributor instructions:

```
git clone --recurse-submodules https://github.com/oyvindberg/bleep.git
cd bleep
bleep compile # I have version 0.0.1-M24
> bleep compile
📗 bootstrapped in 7 ms
📗 bloop-rifle: Starting compilation server
📕 Server didn't start
Read log file:
[0.006s][warning][gc,jvmci] Setting EnableJVMCI to false as selected GC does not support JVMCI: z gc
Bloop server PID: 25246
Ignoring SIGINT
Will truncate output file /Users/David.Francoeur/Library/Caches/build.bleep.bleep/bsp-socket/shared-for-jvm-graalvm-java17_22.3.0/output every 5 minutes
Exception in thread "main" java.net.SocketException: Unix domain path too long
        at java.base/sun.nio.ch.UnixDomainSockets.bind0(Native Method)
        at java.base/sun.nio.ch.UnixDomainSockets.bind(UnixDomainSockets.java:113)
        at java.base/sun.nio.ch.ServerSocketChannelImpl.unixBind(ServerSocketChannelImpl.java:319)
        at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:292)
        at java.base/java.nio.channels.ServerSocketChannel.bind(ServerSocketChannel.java:224)
        at libdaemonjvm.internal.Java16SocketHandler$.server(Java16SocketHandler.scala:41)
        at libdaemonjvm.internal.SocketHandler$.server(SocketHandler.scala:20)
        at bloop.Bloop$.$anonfun$startServer$1(Bloop.scala:167)
        at scala.util.Either.map(Either.scala:353)
        at bloop.Bloop$.startServer(Bloop.scala:165)
        at bloop.Bloop$.$anonfun$main$4(Bloop.scala:89)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
        at libdaemonjvm.server.Lock$.unsafeWritePidAndSetup$1(Lock.scala:32)
        at libdaemonjvm.server.Lock$.$anonfun$tryAcquire$3(Lock.scala:52)
        at scala.util.Either.flatMap(Either.scala:341)
        at libdaemonjvm.server.Lock$.$anonfun$tryAcquire$2(Lock.scala:51)
        at scala.util.Either.flatMap(Either.scala:341)
        at libdaemonjvm.server.Lock$.unsafeCleanUpAndSetup$1(Lock.scala:50)
        at libdaemonjvm.server.Lock$.$anonfun$tryAcquire$9(Lock.scala:100)
        at libdaemonjvm.LockFiles.withLock(LockFiles.scala:37)
        at libdaemonjvm.server.Lock$.tryAcquire(Lock.scala:97)
        at bloop.Bloop$.main(Bloop.scala:89)
        at bloop.Bloop.main(Bloop.scala)
```

I think this is related to the JVM used. When switching to `temurin: 1.17` it worked fine.

**Comment by oyvindberg** [comment]
Hi David, thanks for you interest! :)

One of the things bleep does now is that it starts a bloop server with your chosen JVM. This was in theory optimal since, well, you would compile with your chosen JVM. It does not work in practice though, since bloop doesn't run on any older JVMs :)

I also think it should never use the `system` JVM for this either, but rather fall back on something very standard and useful like say `temurin: 1.17`

**Comment by daddykotex** [comment]
Thanks again for your hard work Øyvind

I believe the issue at hand is just a sad side-effect of https://unix.stackexchange.com/questions/367008/why-is-socket-path-length-limited-to-a-hundred-chars

TLDR: the path used for the socket file given to Bloop is too long on my computer.

This happen because my username is somewhat long: `David.Francoeur` which generates the following folder in to create sockets in: `/Users/David.Francoeur/Library/Caches/build.bleep.bleep/bsp-socket/shared-for-jvm-graalvm-java17_22.3.0/`. This directory is already at 104 characters.

1. I suspect that with shorter username, I would not have an issue
2. We can probably remove one of the `bleep` in `build.bleep.bleep`? maybe?, why two? ([src](https://github.com/oyvindberg/bleep/blob/master/bleep-core/src/scala/bleep/UserPaths.scala#L16))
3. Best action is probably to go into a a shorter directory like `/tmp` for sockets

**Comment by daddykotex** [comment]
TIL about https://github.com/dirs-dev/directories-jvm

Which is shaded by coursier and used here.

**Comment by oyvindberg** [comment]
> I think this is related to the JVM used. When switching to temurin: 1.17 it worked fine.

When you wrote this, did you mean that the warning went away, but it still failed with too long socket name?

**Comment by oyvindberg** [comment]
So well, this socket name limit was completely unknown to me. We can certainly shorten the bleep part of this though:

`build.bleep.bleep/bsp-socket/shared-for-jvm-graalvm-java17_22.3.0/` => 
`build.bleep.bleep/socket-graalvm-java17_22.3.0`
would save a few chars.

I really want to lose that extra bleep in the path name as well, but it just came out of directories-jvm or whatever was used in the end there. feel free to have a look and see if you find a way

**Comment by sideeffffect** [comment]
`build.bleep.bleep` comes from `build.bleep` being the vendor's domain and `bleep` being the name of the application. You can imagine that there might be different applications from `build.bleep` besides `bleep` itself...
